addons/website_sale/views/templates.xml

```diff
@@ -882,6 +867,47 @@
         </div>
     </template>
 
+    <template
+        id="website_sale.products_categories_list_collapsed"
+        inherit_id="website_sale.products_categories_list"
+        name="Collapsed eCommerce Categories"
+    >
+        <xpath expr="." position="replace">
+            <div t-attf-class="{{not isOffcanvas and 'accordion-item'}}">
+                <h6 t-attf-class="accordion-header {{_titleClasses}}">
+                    <button
+                        class="accordion-button px-0 bg-transparent shadow-none"
+                        type="button"
+                        data-bs-toggle="collapse"
+                        t-attf-data-bs-target="#o_wsale_categories"
+                        t-attf-aria-controls="o_wsale_categories"
+                        aria-expanded="true"
+                    >
+                        <b class="o_products_attributes_title">Categories</b>
+                    </button>
+                </h6>
+                <div
+                    t-attf-id="{{not isOffcanvas and 'o_wsale_categories'}}"
+                    t-attf-class="wsale_products_categories_list {{not isOffcanvas and 'accordion-collapse collapse show'}}"
+                >
+                    <ul t-attf-class="nav d-flex flex-column {{'mt-3' if isOffcanvas else 'mb-3'}}">
+                        <li class="mb-1">
+                            <a
+                                t-att-href="keep('/shop', category=0)"
+                                t-attf-class="{{not category and 'text-decoration-underline'}} p-0"
+                            >
+                                All Products
+                            </a>
+                        </li>
+                        <t t-foreach="categories" t-as="c">
+                            <t t-call="website_sale.categories_recursive"/>
+                        </t>
+                    </ul>
+                </div>
+            </div>
+        </xpath>
+    </template>
```

Comment:
That being said, I fear we'll have to deal with this forced complexity in so many places that it might not be worth spending time making this single template clearer :pensive: